lib/scheduler.py

```diff
@@ -339,15 +339,18 @@ def _task(self, name, obj, by, source, dest, value):
             sh = self._sh  # noqa
             try:
                 exec(obj.bytecode)
+                for method in logic.get_method_triggers():
```

Comment:
In general it would be better to only trigger the methods when the logic was successfuly executed.

Using `exit()` is IMHO a bad idea since it should only be used to completely exit the program. A logic is only a piece of code running inside SmartHome and should never use `exit()`.

But if the others like to work around this and catching the exceptions before triggering the methods I can change this.